src/duk_debugger.c

```diff
@@ -37,8 +37,23 @@ DUK_LOCAL void duk__debug_do_detach1(duk_heap *heap) {
 	 * inside the detached callback.
 	 */
 
+	if (heap->dbg_detaching) {
+		return;
+	}
+
 	DUK_D(DUK_DPRINT("debugger transport detaching, marking transport broken"));
 
+	heap->dbg_detaching = 1;
+
+	/* avoid sending multiple Detach notifys */
+	if (heap->dbg_write_cb != NULL) {
+		duk_hthread *thr = heap->heap_thread;
+
```

Comment:
Humm, this was one thing blocking this earlier. I guess using heap_thread would be OK here, the thread doesn't have much bearing on the writing process. But maybe assert thr != NULL here as we're certainly requiring that.